### DISCUSSION TITLE
Konfiguration OpenDTU + Zendure SolarFlow mit Nulleinspeisung

### DISCUSSION THREAD

**Discussion by apss-fuchs** [hoylabs/OpenDTU-OnBattery#1005] [General]
Hier eine kleine Anleitung wie ich das ganze an einem Zendure System betreibe, da der CT-Modus in der Zendure App das nicht so gut regelt.
Es würde natürlich auch mit anderen Batteriespeicher-System funktionieren, da heir der Wechselrichter die Begrenzung vorgibt.

Evtl. gibt es hier auch noch Optimierungsbedarf. Anregungen gerne erwünscht.

**Ich benutze folgende Komponenten:**

- Zendure SolarFlow HUB 2000 + AB 2000 (Batterie)
- Hoymiles HMS-2000-4T Wechselrichter
- 5x 400wp Solar Module (4 am Zendure und 1 direkt am WR)
- Shelly Pro 3EM
- OpenDTU-OnBattery 

**OpenDTU-OnBattery Version**

SDK-Version v4.4.6-dirty
Konfigurationsversion 0.1.28
Firmwareversion 2024.05.07

**OpenDTU Einstellungen:**

1.  **Grundeinrichtung OpenDTU-OnBattery**
Nach Anleitung https://www.opendtu.solar/

2. **Wechselrichter anlegen**
Im Menü unter "Wechselrichter" den Wechselrichter mit der Seriennummer eintragen und schauen ob er sich damit verbinden.
Ansonsten im Menü unter "DTU" die "CMT2300A Sendeleistung" und/oder "CMT2300A Frequenz" anpassen.
**(Wichtig: Es darf keine Hoymiles DTU-Lite o.ä. parallel laufen)
(Hinweis: Es kann bis zu 15 Minuten dauern bis die OpenDTU den Wechselrichter gefunden hat)**

3. **Stromzähler einrichten (Shelly Pro 3EM)**

- Stromzählertyp: HTTP(S) + JSON
- Individuelle HTTP requests pro Phase: Nein
- URL: http://[IP vom Shelly]/rpc/shelly.getstatus
- Authorization Type: None
- Timeout: 2000
- JSON Pfad: em:0/total_act_power
- Einheit: W

_(Weitere Infos: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Power-Meter)_

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/95090185/c032b6d3-450a-4ef9-b22f-e84d438a849b)

4. **Dynamic Power Limiter einstellen**

- Angestrebter Netzbezug: -15W
- Hysterese: 15W
- Wechselrichter wird von Solarmodulen gespeist: Ja
- Minimales Leistungslimit: 120W (4*30W)
- Grundlast: 150W
- Maximales Leistungslimit: 800W
- Stromzählermessung beinhaltet Wechselrichterleistung: Ja

(Weitere Infos: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Dynamic-Power-Limiter)

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/95090185/8e1abdb5-a226-40f8-ae67-b28a7b1d8f84)

**Zendure Einstellungen:**

- Bypass Mode: Always-Off
- Chargelimit: 100%
- Dischargelimit: 15% (Wegen den momentanen Entladeknick)
- PV Inverter Input: 800W MAX
- Energy Plan: Appointment mode (08:00 800W/20:00 150W)

_Grundverbrauch bei mir sind immer min. 150W._

**Comment by Sweti934** [comment]
Danke für die Anleitung! Ginge das auch ohne Shelly wenn ich anderweitig per json o.ä. Die Daten bereit stelle? Du nimmst dir da den aktuellen Leistungswert (gesamt): -200W wären dann Einspeisung korrekt ? Lieben Dank !!

Welche Module hast du genau? Würde mir dann dein System nachbauen wollen

**Comment by apss-fuchs** [comment]
@Sweti934 
Ja das würde auch mit anderen Verbrauchszählern gehen die eine Schnittstelle (JSON oder MQTT) bereitstellen.

Ich habe die Trina Solar Vertex S 400 Wp Module

**Reply by Sweti934** [reply]
Mega cool vielen Dank! Bist du noch zufrieden damit und klappt es soweit mit der Regelung? Echt Mega gut!!

**Reply by apss-fuchs** [reply]
Ja funktioniert sehr gut damit.

**Comment by stegbth** [comment]
Hi @apss-fuchs ,

der Shelly Pro 3EM scheint ja unterschiedliche Abfrage URLs zu haben.

bei dir lautet der
`http://[IP vom Shelly Pro 3EM]/rpc/shelly.getstatus`
`JSON Pfad: em:0/total_act_power`

bei mir habe ich nach einigem Suchen folgendes angegeben.
`http://[IP vom Shelly Pro 3EM]/rpc/EM.GetStatus?id=0`
`JSON Pfad: total_act_power`


Wo liegt hier der Unterschied?

vg
Thomas

**Reply by apss-fuchs** [reply]
Hi @stegbth,

für diesen Einsatzzweck gibt es keinen Unterschied.
"shelly.getstatus" liefert alle Werte und "EM.GetStatus?id=0" nur die Daten von EM:0

**Comment by edddymax** [comment]
Danke für die Anleitung. Habe 3 Änderungen : Nur eine Einstellung im Terminmodus (falls mal wieder der Server von Zendure aussteigt, würden sich die Werte nicht ändern. Muss ja auch nicht, openDTU regelt); Nutze IR Abfrage mit Tasmota (anstelle des Shelly, geht auch schneller und ist lokal); Habe zwei Hubs am HMS 1600 4T (schnelle Steuerung und gleichmäßige Verteilung der Leistung)
Läuft seit 3 Tagen sehr stabil . Die Zendure-App nutze ich nur noch zur Kontrolle der Ackuleistung ( Füllstände).

**Reply by leo-3889** [reply]
Hallo, ich habe das exakt gleiche Setup: Tasmota IR Reader, Opendtu-onbattery, HMS1600. Ich überlege, vielleicht ein AIO2400 hinzuzufügen... Würden Sie Ihr Setup empfehlen? Haben Sie ein Panel auf dem HMS, das unabhängig von der Batterie ist? Vielen Dank!

**Comment by sweti3** [comment]
Super danke! Könntet ihr eure HW Verschaltung mal darstellen? Wie funktioniert das mit 2 Hubs an einem HMS? Die Panels gehen an den Hub und von dort an den HMS? Wäre dankbar für ein kleines "Diagramm" @edddymax und @apss-fuchs  - ganz lieben Dank schonmal!

**Reply by apss-fuchs** [reply]
Es sind 4 Panels (jeweils 2 Panels mit Y-Kabel pro Eingang) an dem Hub 2000 angeschlossen.
Am Hub Ausgang geht es dann wieder mit Y-Kabel an 2 Eingänge des Wechselrichters.

**Reply by edddymax** [reply]
PVModule (je 2 parallel) — Hub 2000  —(2fach Kabel) — WR (PV1 und 2)

PVModule                      — Hub 1200 —(2fach Kabel) — WR (PV3 und 4)

**Comment by Janorei** [comment]
Ich habe ein sehr ähnliches System aufgebaut, das auch zu vollster Zufriedenheit läuft Unterschiede:

-     HM 800 WR
-     Bypass Mode: Automatisch
-     Energy Plan: keiner
wie @edddymax benutze ich die zendure App nur ab und an zum Schauen, welche Leistung momentan wo hin geht.
Ansonsten habe ich alle Werte vom Zendure System per MQTT in meinem Homeassistant eingebunden.

**Comment by haeff123** [comment]
Hallo. Ich habe derzeit OpenDTU OB mit Shelly Peo 3EM ohne Speicher laufen. Funktioniert einwandfrei. Jetzt will ich einen Hub 2000 kaufen und verstehe ich das richtig - bitte nicht lachen - ich kann dann am Hub eine Ausgangsleistung z.b. 800 Watt einstellen - entscheidend für die Entladung des Akku ist aber dann das Limit des Wechselrichters ?! Im extremen Fall stelle ich den WR auf 100Watt und dann ist genau dieser Wert entscheidend für die Entladung des Akku ...  Verstehe ich das so richtig ?
Grüße

**Reply by Janorei** [reply]
Ja, genau so wird das gesteuert, ich habe eine solche Anlage seit Monaten laufen (Hub 2000 + 2 AB2000). Open DTU regelt den WR und der Hub gibt nur soviel raus, weil der WR möchte, der Rest geht automatisch in den Akku. Bitte am Hub keinerlei Energiemodus einstellen und Bypass auf automatisch lassen. Funktioniert perfekt, ich bin jeden Tag (und Nacht) begeistert. Durch die Open DTU Steuerung hast Du auch die ganzen Probleme der z.T. Noch nicht ganz ausgereiften Software von Zendure nicht, die die Benutzer in der Zendure FB Gruppe immer melden

**Reply by haeff123** [reply]
OK. Klingt ja top. Smart-CT Modus ist ja nicht so performant wie man liest und die OpenDTU Box funktioniert wirklich tadellos.  Danke fürs Feedback.

**Reply by Janorei** [reply]
Ein kleiner Nachtrag: Manchmal gibt der Hub nicht die Leistung raus, die der WR möchte, sondern weniger. Das liegt dann an der nicht ganz gut laufenden Bypass Implementierung im HUB. Die kannst Du nur beeinflussen, wenn Du den Bypass ganz ausschaltest, dann geht aber alles über den Akku (ist aber nicht besonders gut für den Akku). Ich kann ganz gut mit dem ersteren Manko leben

**Reply by haeff123** [reply]
Was für ein Einsparpotential hast du bei deinem Setup ?

**Reply by Janorei** [reply]
Ein wenig schwierig zu beurteilen, weil das nicht meine einzige Energiequelle ist

**Reply by strelok312** [reply]
> Ein kleiner Nachtrag: Manchmal gibt der Hub nicht die Leistung raus, die der WR möchte, sondern weniger. Das liegt dann an der nicht ganz gut laufenden Bypass Implementierung im HUB. Die kannst Du nur beeinflussen, wenn Du den Bypass ganz ausschaltest, dann geht aber alles über den Akku (ist aber nicht besonders gut für den Akku). Ich kann ganz gut mit dem ersteren Manko leben

Noch eine kurze Frage dazu: kommt dieses Verhalten nur dann zum Tragen wenn der Akku voll ist? Oder auch in der Ladephase? Und Du hast den Bypass komplett ausgemacht, richtig?

**Reply by Janorei** [reply]
Nein ich hsbe bypass auf automatisch und bei mir funktioniert das gut

**Comment by Diddes** [comment]
Hallo , bin neu hier . Habe jetzt alles so gemacht wie oben beschrieben aber weiter als http anfrage wird gesendet kommt nichts . 
Muss halt sagen das ist alles neuland für mich .

**Reply by apss-fuchs** [reply]
Wo genau hängst du?

**Reply by Diddes** [reply]
Er verbindet sich nicht mit Shelly . Es steht als Verbindung würde hergestellt http und es ist auch grün hinterlegt

**Comment by Gnoesch** [comment]
Hallo zusammen,
ich habe heute OpenDTU on battery zusammen mit meinem Hub2000, 2xAB2000 und dem Shelly Pro 3EM in Betrieb genommen. Ich habe jetzt alles soweit eingerichtet, allerdings stelle ich mir gerade die Frage, wie weiß der Hub2000 denn was der WR jetzt an Leistung benötigt? Vielleicht stehe ich irgendwie auf dem Schlauch oder habt ihr den Shelly auch noch in der Zendure App an den Hub konnektiert.
Viele Grüße
Fred

**Reply by Janorei** [reply]
Nein, den Shelly auf keinen Fall in die Zendure Landschaft einbinden. Am Zendure Hub in der App keinerlei Energiemodus auswählen und als Ausgangsleistung HUB --> Haus in der App die maximal Leistung Deines WR einstellen (vermutlich 800 W) . Da der WR ja über die Open DTU sein Limit eingeregelt bekommt bezieht er auch maximal genau die momentan benötigte Leistung die er möchte vom HUB. Bypassmodus in der App vom HUB habe ich persönlich auf "automatisch" gelassen.

**Reply by strelok312** [reply]
Hallo, bin ganz neu hier und habe mein System gestern in Betrieb genommen, gleiche Anlage aber bisher nur mit einem AB2000. Der Shelly war bei mir im Zendure Account und App eingebunden weil das bisher ohne  openDTU im CT Modus lief. Ich habe alles so eigestellt wir hier beschrieben, allerdings habe ich den Shelly bei Zendure eigebunden gelassen. Dadurch dass KEIN Energiemodus gewählt ist wird Shelly in der Zendure App nur zur Darstellung des Hausverbrauchs gebraucht - nach meinem Verständnis. 
Leistungsabgabe wird von meinem HM1200 über opendDTU-OnBattery geregelt. Ich würde sagen man kann den Shelly eingebunden lassen, zumindest funktioniert es so bei mir.

**Reply by Janorei** [reply]
Okay, wieder was gelernt, wenn der Shelly in Zendure drinbelieben kann auch o.k. Ich hatte ihn halt nie eingebunden.

**Reply by strelok312** [reply]
Wie habt ihr eigentlich eure Visualisierung/Dashboard zur Ertragsübersicht organisiert? Deshalb habe ich den Shelly auch in ZendureApp dringelassen, damit ich diese Zusammenhänge zwischen Ladeleistung/Akku/Hausverbrauch sehen kann. Trotzdem bin auf der Suche nach einer schönen anderweitigen Visualisierung, habe mich aber noch nicht an HomeAssistant via Mosquitto getraut. Gibt es denn einen einfachere Möglichkeit Ertragsdaten zu sehen? Auf dem Board Livedaten der OpenDTU-OnBattery sind mir zu wenig Informationen drauf. Was gibt es hier noch für einfache und zugängliche Möglichkeiten? Danke und Gruß.

**Reply by Janorei** [reply]
Ich habe es im Homeassistant visualisiert und ja, es ist Arbeit, selbst wenn man wie ich aus der Programmiererecke kommt

**Reply by Gnoesch** [reply]
Guten Morgen zusammen,

mir fällt vermehrt auf, dass der WR sehr häufig bei -50 oder -85Watt ins Nett einspeist.

![IMG_6693](https://github.com/user-attachments/assets/c9e23959-187a-46f5-81b8-92ae2ec588f7)

Ich habe mittlerweile den max Einspeisewert für den WR auf 300W eingestellt, da wir einen relativ hohen Grundverbrauch haben. Habt ihr eine Idee woran es liegen könnte? Ich kann gerne auch Screenshots meiner openDTU on battery Einstellungen schicken.

**Reply by codenyte** [reply]
> Ich habe es im Homeassistant visualisiert und ja, es ist Arbeit, selbst wenn man wie ich aus der Programmiererecke kommt

@Janorei Könntest du vielleicht deine Home Assistant Konfiguration, zumindest den Teil, der für Zendure/OpenDTU etc. zuständig ist, teilen? Vielen Dank im Voraus.

**Comment by multiball200** [comment]
Sehr interessante Config, wäre genau nach meinen Anforderungen mit den 5 Modulen.
Wird das einzelne Modul dann mit Y-Verteilern an einen String des WR angeschlossen ? Also an 2x2 Buchsen und die anderen 4 dann an dem Zendure Hub ? Habe 2x Module Süd 405W und 2x410W West. Würde dann gerne noch ein Modul vermutlich Süd benutzen.
Der HMS hat ja sicher 4 Eingänge, kann auch ein anderer Hoymiles zb. HMS 1600 ohne WLAN benutzt werden, und welche DTU on Batt. hast Du ? Aber letztendlich kommt es halt drauf an das ich auf die max. zulässigen 800W komme die eingespeist werden.

**Comment by M4tt3s** [comment]
Ich fahre ebenfalls ein BKW mit einem Zendure Solarflowsystem in Verbindung mit einer openDTU on Battery und einem Shelly Pro3EM. 
Mit der o.g Beschreibung hatte der Zendur " CT-Modus" für mich ausgedient und alles lief perfekt, bis ich den Shelly geupdatet habe.

Die aktuelle Shelly FW " 2020726-114514/1.4.0-gb2aeadb " lässt die openDTU den JSON Pfad: em:0/total_act_power anscheinend nicht mehr auslesen (oder hat sich geändert) und ich erhalte von der DTU Fehlermeldungen beim Speichern der Einstellungen. 

Seitdem läd der WR sporadisch um die 50W ins Haus.

Kurz gefasst: Hat jemand seit dem aktuellen Shelly FW-Update ein Problem zwischen DTU und Pro3EM

Vielen Dank!

**Reply by M4tt3s** [reply]
Fehlermeldung in der DTU lautet wie folgt:

[HttpPowerMeter] Unable to access JSON key 'total_act_power' (JSON path 'em:0/total_act_power', position 5)

**Reply by spcqike** [reply]
Öffne die URL doch mal im Browser. Wie sieht die json Antwort den nach dem Update aus?

**Reply by M4tt3s** [reply]
Danke für den Tipp! Also die Antwort sah wie folgt aus, dass sämtliche werte im Code mit  "null" ausgegeben wurden.
Ich habe kurzer Hand und Verdacht den 3ProEM einem Werksreset unterzogen und siehe da, OpenDTU bekomt wieder den entsprechenden Wert zurück.

**Comment by dk7td** [comment]
Hallo,

und danke für die tolle Beschreibung! Eine Frage hätte ich allerdings. Was genau bewirkt diese Einstellung:

>     * Wechselrichter wird von Solarmodulen gespeist: Ja

Wenn ich den WR am Zendure-Hub habe, dann hängt er ja nicht direkt an den Solarmodulen.

Danke!

**Comment by flatty123** [comment]
Moin,danke für die super Anleitung!
Es läuft super schnell, nur irgendwie habe ich es wohl noch nicht richtig eingestellt. 
Meine gesamte Leistung des HMS800 (ohne Limit) bleibt bei 630W Zendure-App und 660W bei DTU stehen, 
obwohl maximales Leistungslimit: 800W eingestellt ist.
Bei der Zendure-Schneckenpost ging es vorher locker nach 6-10Minuten :-) bis 820W?
![dtu](https://github.com/user-attachments/assets/b051c41f-c7c8-45a5-b703-8e28a068b30b)

**Reply by SolarFreund** [reply]
@apss-fuchs:  danke für die Darstellung. Nach viel Recherche bin ich theoretisch auch bei deinem  Set gelandet. 
Hub2000 wegen der besseren Leistung 1200 Watt zum WR. 

Dann habe ich inzwischen auch vom dauernden Ärger  und unterirdischen Regelzeiten gelesen. Ich brauche  was stabiles !!!  Konfigurieren und laufen lassen. 
So will ich 2x Noah2000 nehmen  und gut. (Das sind 4 kWh für 250 € mehr)

Übersehe ich da was?

Frage nebenbei:
Da bei ca. 1800 Wp schnell Überschuss-PV da ist:  wo merkt OpenDtu das und schaltet einen Verbraucher (heizstab im Boiler) ein?

Danke an alle Tippgeber.

**Reply by spcqike** [reply]
> wo merkt OpenDtu das und schaltet einen Verbraucher (heizstab im Boiler) ein?

gar nicht. Das ist Aufgabe eines EMS. 

Den einzigen „Verbraucher“ den die openDTU einstellen kann, ist ein regelbares Netzteil zum aufladen eines Akkus.

**Comment by masterblaster15** [comment]
Hallo, ich habe genau diese Konfiguration. 
Es passt auch alles super, die Einstellungen habe ich auch genau so gemacht. 
Jedoch geht der HUB2000 sobald die Akkus denn Mindeststand erreicht haben 
Offline. Woran kann das liegen? Oder soll das so sein?

**Reply by henkel1976** [reply]
Hey. Ich habe mein Zendure auch mit Open DTU on Battery laufen. Schau mal in der Zendure App bei den Akkueinstellungen, da steht bei dir bestimmt "Abschaltung" stell das ganze mal auf "Entladen beenden"

**Reply by dk7td** [reply]
Du kannst das hier einstellen: entweder „Shutdown“ oder „Stop Discharging“.

<img width="744" alt="image" src="https://github.com/user-attachments/assets/18e587e6-2b97-4a75-8b39-a2db62f9f557">

**Reply by masterblaster15** [reply]
Danke, genau das war es. Das muss dann beim Update vom 
HUB2000 passiert sein. Denn sonst war bei mir immer Entladen 
beenden drin. Ich Danke noch mal, auch für die super Anleitung.

**Reply by codenyte** [reply]
Hat es irgendwelche Vor- oder Nachteile, den Zendure Hub auszuschalten oder anzulassen? Im Prinzip sollte das doch keinen Unterschied machen, oder?

**Reply by dk7td** [reply]
In der dunklen Jahreszeit erspart man sich damit den Eigenverbrauch des Hubs. Der schaltet dann bei den ersten Sonnenstrahlen wieder ein.

**Reply by codenyte** [reply]
Danke, dann werde ich das wohl so einstellen.

**Reply by dk7td** [reply]
Nachteil ist, daß der Hub im Shutdown nicht erreichbar ist, falls man ihn bedienen oder konfigurieren möchte. Daher verwende ich diesen Modus nur, wenn es nicht anders geht.
Im Sommer habe ich als Entladegrenze 15% eingestellt. Über den Winter werde ich damit höher gehen. Wenn längere Zeit wenig bis keine Solarleistung reinkommt, kann der Shutdown-Modus schon sinnvoll sein. Z.B. wenn Schnee auf den Modulen liegt. Dann zieht der Eigenverbrauch den Akku irgendwann leer. Der Modus wurde damals eingeführt, als die Akkus im Winter wie die Fliegen „gestorben“ sind, weil die User sie zu tief entladen haben.

**Comment by Josef1966** [comment]
Hallo, ich habe vergleichbare Hardware, abweichend nur den HMS-1600-4T und 4 Module.
Alle 4 Module hängen parallel an den 2 Eingängen des Hubs. Den Hub-Ausgang verteile ich auf alle 4 Eingänge des WR.
In der Zendure-App alle Energiepläne deaktiviert, Steuerung komplett über OpenDTU oB.
Wenn nachts der Akku entladen wird, werden da aber nur ca. 20-40W vom WR ins Hausnetz ausgegeben, obwohl der Verbrauch deutlich höher ist.
An was kann das liegen? Sollte man besser nur 2 WR-Einhänge anstelle von 4 belegen? Wenn ja, welche dann, 1/2 oder 3/4 oder 1/3 oder 2/4?

**Comment by Lala1178** [comment]
Hallo zusammen.

Ich bin ein absoluter Neuling in dieser Sache und habe mich mit dem Thema seit kurzem befasst.

Grundsätzlich habe ich die gleiche Hardware wie der Verfasser des Artikels und wollte jetzt auch die Nulleinspeisung über die DTU machen und nicht mehr über die Zendure App.

Die DTUonBattery habe ich mir im DTUshop bestellt und schon in meinem Heimnetzwerk eingerichtet und den Wechselrichter habe ich unter Einstellung in der DTU Software angegeben. Das hat auch soweit alles funktioniert das ich ihn dort sehe und er mir aktuelle Werte vom Wechselrichter übermittelt.

Jetzt kommt mein Problem!!! Ich finde auf der DTU Seite nicht die Einstellmöglichkeit für Stromzähler bzw. Dynamic Power Limiter. Muss ich dazu noch was installieren oder was mache ich da falsch!? Vielleicht bin ich ja auch nur blind!!!


Vielleicht gibt es ja hier einen Experten der einem Laien weiter helfen kann.

Danke schonmal im voraus.

**Comment by Harisjems** [comment]
Du hast wahrscheinlich nur OpenDTU Firmware; die unterscheiden sich optisch nicht viel, aber OpenDTU OnBattery hat mehr Funktionen, wie du festgestellt hast.  Das heißt für dich, du müsstest OpenDTU OnBattery Neu flashen. Am besten die Factory-Firmware aus dem Download-Bereich mit PC und ESP-Download-Tool.  Vorher aber bitte das Pin-Mapping sichern (Anschluss-Einstellung): Display, NRF24, CMT-Modul. Unter Einstellungen/Konfigurationsverwaltung. Und dann nach dem Flashen wieder einspielen und aktivieren.
![Screenshot_20250418_191825_Samsung Internet](https://github.com/user-attachments/assets/ca425549-f3b9-4363-9191-1f51e652979d)

**Comment by Lala1178** [comment]
Könntest du mir noch bitte eine Link hier schreiben wo ich die Software finde!? Das wäre sehr nett.

**Reply by Harisjems** [reply]
Alles hier zum Finden :-) Und dann die passende Version für deinen ESP-Chip halt.
https://github.com/hoylabs/OpenDTU-OnBattery/actions/runs/14519186399
![Screenshot_20250418_201111_Samsung Internet](https://github.com/user-attachments/assets/53d4ec9f-9d29-400a-855e-446dc7c91e62)
![Screenshot_20250418_201335_Samsung Internet](https://github.com/user-attachments/assets/f7f568a4-23d4-4cfe-8275-279ccfa6393c)

**Reply by Lala1178** [reply]
Irgendwas mache ich immer noch falsch ;-(. Von deinem Link habe ich mir die Software runter geladen (hatte geschaut was ich bis jetzt hatte: Chip Modell EWSP32-S3 und Firmeware: generic_esp32s3_usb). Unter Einstellung gibt es ja auch Firmeware Aktualisierung auf der DTU Seite. Das habe ich durchgeführt mit der Datei "opendtu-onbattery-generic_esp32s3_usb.factory.bin". Das lief auch ohne Probleme durch und er hat gesagt das er die DTU neu Starten wird. Danach habe ich aber weiterhin die gleiche Aufmachung und weiterhin kann ich die gewünschten Reiter für Stromzähler und Dy. Power Limiter nicht finden.

Was mach ich nur falsch ;-(!!??

![image](https://github.com/user-attachments/assets/66847993-ea2d-46c8-aec6-7fff5cdb3ed6)

**Reply by Harisjems** [reply]
Du hast die falsche Firmware gewählt.  Factory kann man nur über PC und ESP Download Tool flashen. Du musst für das OTA Update die Datei generic_esp32s3_usb.bin 
nehmen (ohne Factory).
VORHER Natürlich entpacke.
![Screenshot_20250419_122842_RAR](https://github.com/user-attachments/assets/130126c6-8550-427f-b628-51be264f1bd4)




 Es wird zwar nicht empfohlen, aber es hat bei mir funktioniert. Ich weiß nicht, z. B., wie sich aktuelle openDTU und aktuelle openDTUonBattery Partitionen unterscheiden.
Das wird zb.bei factory. bin Automatich mit erledigt ( neu Partitioniert/ Flashen) aber nur mit Zusatzsoftware an PC oder (Web-Installer was nicht funktioniert:-) )
  Alles ohne Schiess Gewähr:-). Pin-Mapping unbedingt vorher sichern. Mfg

**Reply by Lala1178** [reply]
Vielen vielen Dank!!!

Es hat endlich funktioniert. Ohne deine Hilfe hätte ich das nicht hinbekommen.

Also Danke nochmal und noch einen schönen Abend.

**Reply by Lala1178** [reply]
Guten Abend. 

Ich habe es jetzt geschafft eine Verbindung zu meinem Shelly herzustellen und ich sehe auch den aktuelle Leistung/Netz. 

Den Dynamic Power Limiter habe ich auch so eingestellt wie oben beschrieben. Gespeichert und er sagt er hat auch alles so übernommen.

Auch in der Zendure App habe ich alles so eingestellt wie beschrieben.

Aber ich sehe nicht das er auf eine Nulleinspeisung regelt. Ich habe das Gefühl der Dynamic Power Limiter Arbeit nicht. 

Kennt jemand das Problem bzw. Hat eine Lösung dafür!?

**Reply by Josef1966** [reply]
Vergleiche doch mal deine Einstellungen mit diesen:

https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Zendure-SolarFlow-with-Hoymiles-Inverter

**Reply by Lala1178** [reply]
Hallo.

Einen Tag später hat alles funktioniert. Wobei ich nicht weis was ich anders gemacht habe. Aber egal Hauptsache es funktioniert.

Danke nochmal an alle die mir mit ihrem Wissen weiter geholfen haben.

**Comment by guenni26** [comment]
> Hier eine kleine Anleitung wie ich das ganze an einem Zendure System betreibe, da der CT-Modus in der Zendure App das nicht so gut regelt. Es würde natürlich auch mit anderen Batteriespeicher-System funktionieren, da heir der Wechselrichter die Begrenzung vorgibt.
> 
> Evtl. gibt es hier auch noch Optimierungsbedarf. Anregungen gerne erwünscht.
> 
> **Ich benutze folgende Komponenten:**
> 
> * Zendure SolarFlow HUB 2000 + AB 2000 (Batterie)
> * Hoymiles HMS-2000-4T Wechselrichter
> * 5x 400wp Solar Module (4 am Zendure und 1 direkt am WR)
> * Shelly Pro 3EM
> * OpenDTU-OnBattery
> 
> **OpenDTU Einstellungen:**
> 
> 1. **Grundeinrichtung OpenDTU-OnBattery**
>    Nach Anleitung https://www.opendtu.solar/
> 2. **Wechselrichter anlegen**
>    Im Menü unter "Wechselrichter" den Wechselrichter mit der Seriennummer eintragen und schauen ob er sich damit verbinden.
>    Ansonsten im Menü unter "DTU" die "CMT2300A Sendeleistung" und/oder "CMT2300A Frequenz" anpassen.
>    **(Wichtig: Es darf keine Hoymiles DTU-Lite o.ä. parallel laufen)
>    (Hinweis: Es kann bis zu 15 Minuten dauern bis die OpenDTU den Wechselrichter gefunden hat)**
> 3. **Stromzähler einrichten (Shelly Pro 3EM)**
> 
> * Stromzählertyp: HTTP(S) + JSON
> * Individuelle HTTP requests pro Phase: Nein
> * URL: http://[IP vom Shelly]/rpc/shelly.getstatus
> * Authorization Type: None
> * Timeout: 2000
> * JSON Pfad: em:0/total_act_power
> * Einheit: W
> 
> _(Weitere Infos: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Power-Meter)_
> 
> ![grafik](https://private-user-images.githubusercontent.com/95090185/333803355-c032b6d3-450a-4ef9-b22f-e84d438a849b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwMDIxNjIsIm5iZiI6MTc0NTAwMTg2MiwicGF0aCI6Ii85NTA5MDE4NS8zMzM4MDMzNTUtYzAzMmI2ZDMtNDUwYS00ZWY5LWIyMmYtZTg0ZDQzOGE4NDliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDE4NDQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkwZDM2MTI4OGNjZmQ0NDE4MjQ3NjExOWI5OWQ5MTcyMDA5YmNlMjA3M2U4ZmY2MTkwMmU4NTVmYTRkMWM3MTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WJDE9VJzwnE6cPBJSZvzJWUAAMsxH0qUtFxm7rfsDjw)
> 
> 4. **Dynamic Power Limiter einstellen**
> 
> * Angestrebter Netzbezug: -15W
> * Hysterese: 15W
> * Wechselrichter wird von Solarmodulen gespeist: Ja
> * Minimales Leistungslimit: 120W (4*30W)
> * Grundlast: 150W
> * Maximales Leistungslimit: 800W
> * Stromzählermessung beinhaltet Wechselrichterleistung: Ja
> 
> (Weitere Infos: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Dynamic-Power-Limiter)
> 
> ![grafik](https://private-user-images.githubusercontent.com/95090185/333803496-8e1abdb5-a226-40f8-ae67-b28a7b1d8f84.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwMDIxNjIsIm5iZiI6MTc0NTAwMTg2MiwicGF0aCI6Ii85NTA5MDE4NS8zMzM4MDM0OTYtOGUxYWJkYjUtYTIyNi00MGY4LWFlNjctYjI4YTdiMWQ4Zjg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDE4NDQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdlN2UwNGZhOTU1MGU3ZmJlMDA1MDU5Mjk2Y2U1ZThjYTNjNzVjNWRhNzBmZGNkYmYwNTQ4YzQwNzc0Zjg4MjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1F3S1CesaeyhseO3WpbUkMj1syiOW5tmMeoe2U0eSZ0)
> 
> **Zendure Einstellungen:**
> 
> * Bypass Mode: Always-Off
> * Chargelimit: 100%
> * Dischargelimit: 15% (Wegen den momentanen Entladeknick)
> * PV Inverter Input: 800W MAX
> * Energy Plan: Appointment mode (08:00 800W/20:00 150W)
> 
> _Grundverbrauch bei mir sind immer min. 150W._

Welche DTU SW-Vers. benutzt du?

**Reply by Harisjems** [reply]
Ist es nicht so, dass in neueren OpenDTU-OnBattery-Versionen bei Smartbuffer-Batterien betriebenen WR als Stromquelle immer Smartbuffer gewählt werden sollte, da es sonst zum unerwünschten „Aua“-Effekt kommt?  Habe ich vom Macher mitbekommen; sonst macht der DPL unbezahlte Überstunden.

**Reply by Josef1966** [reply]
siehe auch Wiki:

https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Zendure-SolarFlow-with-Hoymiles-Inverter

**Reply by apss-fuchs** [reply]
Hier meine Version:

SDK-Version v4.4.6-dirty
Konfigurationsversion 0.1.28
Firmwareversion 2024.05.07
